src/parser/shared/modules/items/bfa/dungeons/IgnitionMagesFuse.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS/index';
+import ITEMS from 'common/ITEMS/index';
+import ItemLink from 'common/ItemLink';
+import { formatNumber, formatPercentage } from 'common/format';
+import { calculateSecondaryStatDefault } from 'common/stats';
+import HasteIcon from 'interface/icons/Haste';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import ItemStatistic from 'interface/statistics/ItemStatistic';
+import BoringItemValueText from 'interface/statistics/components/BoringItemValueText';
+import Events from 'parser/core/Events';
+import StatTracker from 'parser/shared/modules/StatTracker';
+
+const ACTIVATION_COOLDOWN = 120; // seconds
+
+/**
+  Ignition Mage's Fuse
+  Item Level 340
+  Binds when picked up
+  Unique-Equipped
+  Trinket	
+  +205 Intellect
+  Use: Ignite the fuse, gaining 168 Haste every 4 sec. Haste is removed after 20 sec. (2 Min Cooldown)
+ */
+
+class IgnitionMagesFuse extends Analyzer {
+  static dependencies = {
+    statTracker: StatTracker,
+  };
+
+  statBuff = 0;
+  uses = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrinket(ITEMS.IGNITION_MAGES_FUSE.id);
+    if(!this.active) {
+      return;
+    }
+
+    this.statBuff = calculateSecondaryStatDefault(340, 164, this.selectedCombatant.getItem(ITEMS.IGNITION_MAGES_FUSE.id).itemLevel);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.IGNITION_MAGES_FUSE_BUFF), this.onUse);
+  }
+
+  onUse(event) {
+    this.uses += 1;
+  }
+
+  get utilization(){
+    return this.uses / this.possibleUseCount;
+  }
+
+  get getAverageHaste(){
+    const averageStacks = this.selectedCombatant.getStackWeightedBuffUptime(SPELLS.IGNITION_MAGES_FUSE_BUFF.id) / this.owner.fightDuration;
+    return averageStacks * this.statBuff;
+  }
+
+  get possibleUseCount() {
+    return Math.ceil(this.owner.fightDuration / (ACTIVATION_COOLDOWN * 1000));
+  }
+
+  get uptime() {
+    return this.selectedCombatant.getBuffUptime(SPELLS.IGNITION_MAGES_FUSE_BUFF.id) / this.owner.fightDuration;
+  }
+
+  statistic() {
+    return (
+      <ItemStatistic
+        size="flexible"
+        tooltip={(
+          <>
+            You activated your Ingntion Mage/'s Fuse <b>{this.uses}</b> of <b>{this.possibleUseCount}</b> possible time{this.uses === 1 ? '' : 's'}.<br/>
+            Buff uptime was {formatPercentage(this.uptime, 0)}%, utilization was {formatPercentage(this.utilization, 0)}%.
+          </>
+        )}
+      >
+        <BoringItemValueText item={ITEMS.IGNITION_MAGES_FUSE}>
+          <HasteIcon /> {formatNumber(this.getAverageHaste)} <small>Average Haste gained</small> <br />
+        </BoringItemValueText>
+      </ItemStatistic>
+    );
+  }
+
+  get suggestedUsage() {
+    return {
+      actual: this.utilization,
+      isLessThan: {
+        minor: .8,
+        average: .7,
+        major: .6,
+      },
+      style: 'number',
+    };
+  }
+  suggestions(when) {
```

Comment:
Might be worth using the inbuilt "cast efficiency" suggestion here instead of creating it for yourself again
example: https://github.com/WoWAnalyzer/WoWAnalyzer/blob/94848149ec7e9b63e8f97f648282006c04ebf392/src/parser/shared/modules/items/bfa/raids/crucibleofstorms/LurkersInsidiousGift.js#L51-L60

Reply:
Thanks for the suggestion, I didn't know that existed.  I'll add it to the PR.

Reply:
I changed to this:
   ```
 this.abilities.add({
      spell: SPELLS.IGNITION_MAGES_FUSE_BUFF,
      buffSpellId: SPELLS.IGNITION_MAGES_FUSE_BUFF.id,
      name: ITEMS.IGNITION_MAGES_FUSE.name,
      category: Abilities.SPELL_CATEGORIES.ITEMS,
      cooldown: ACTIVATION_COOLDOWN,
      gcd: null,
      castEfficiency: {
        suggestion: true,
      },
    });
```
But it's no longer picking up the number of casts correctly, this might be because its a stacking buff, I'm not sure, I double checked the spellBuffId and it is correct.  But on the suggestion it is always showing zero casts.

Is there a way to specify the number of casts?  I have `this.users` returning the correct value, if there is a way to pass it through.

Reply:
it looks like for some reason that specific trinket doesn't have a cast event associated with it, which is why the tracker isn't tracking it by default (might be something worth adding into a normalizer at some point to fix missing "CAST" events)

adding the, albeit deprecated, casts and maxCasts should be enough to make it work
```javascript
this.abilities.add({
      spell: SPELLS.IGNITION_MAGES_FUSE_CAST,
      name: ITEMS.IGNITION_MAGES_FUSE.name,
      category: Abilities.SPELL_CATEGORIES.ITEMS,
      cooldown: ACTIVATION_COOLDOWN,
      castEfficiency: {
        suggestion: true,
        casts: () => this.uses,
        maxCasts: () => this.possibleUseCount,
      },
    });
```

Reply:
I have added `maxCasts: () => this.possibleUseCount` to the PR.  I looked around for an example that shows the fabricating of a cast event.  If someone could point me in the right direction that would be great.
Thanks,